packages/contentful--create-contentful-app/src/getTemplateSource.ts

```diff
@@ -92,6 +92,10 @@ async function makeContentfulExampleSource(options: CLIOptions): Promise<string>
     return selectTemplate(ContentfulExample.Typescript);
   }
 
+  if (options.deliveryFunction) {
+    return selectTemplate(ContentfulExample.Typescript);
+  }
```

Comment:
I'm not sure if I understand the business logic here. What does using delivery functions have to do with template selection (js vs ts)?

Reply:
It is indeed a bit confusing - the problem is that delivery functions are created side by side near the code for the frontend app as well.
We want to give the possibility to omit a `--javascript` or a `--typescript` flag. So if a user passes only the `--delivery-function` flag then we assume it will be Typescript for both delivery function code and the frontend code. I chose Typescript because it is already written in the help that the typescript is a default option.